### PR TITLE
Drop tests of Spikeinterface when `WaveformExtractor` is not available

### PR DESCRIPTION
**Pull request by h-mayorquin**
This will get dev tests working properly where the latest version of Spikeinterface is used.

We will discuss more slowly how we want to move forward in supporting the new way of extracting waveforms in Spikeinterface.

### CODE REVIEW

File: src/neuroconv/tools/spikeinterface/spikeinterface.py [hunk 1]
```diff
@@ -1254,7 +1254,7 @@ def write_sorting(
 
 
 def add_waveforms(
-    waveform_extractor: WaveformExtractor,
+    waveform_extractor,
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/spikeinterface/spikeinterface.py:1257]:
```suggestion
    waveform_extractor: "WaveformExtractor",
```
Should we at least do this?

Comment (h-mayorquin) [reply src/neuroconv/tools/spikeinterface/spikeinterface.py:1257]: Yeah, I think that's better. Let me push this.

Comment (CodyCBakerPhD) [reply src/neuroconv/tools/spikeinterface/spikeinterface.py:1257]: Or else nest entirely in a try/except scope

try:
    from spikeinterface import WaveformExtractor

    def add_waveforms(waveform_extractor: WaveformExtractor, ...)
    ...
except Exception as exception:
    assert type(exception) is <type of exception - I think an ImportError?)
    assert "Some string part of the message unique to the failured import of waveforms extractor" in str(exception)

Comment (CodyCBakerPhD) [reply src/neuroconv/tools/spikeinterface/spikeinterface.py:1257]: Or, or,

LIke you do in tests, nest entirely in `if SI_VERSION > threshold` block?

Comment (h-mayorquin) [reply src/neuroconv/tools/spikeinterface/spikeinterface.py:1257]: I rather just add the docstring. I don't think is worth it with an object that will be dropped anyway.

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**